Authentication/LoginProviders/Google/GoogleLoginProvider.swift

```diff
@@ -0,0 +1,111 @@
+//
+//  GoogleLoginProvider.swift
+//  Authentication
+//
+//  Created by Nahuel Gladstein on 6/28/17.
+//  Copyright © 2017 Wolox. All rights reserved.
+//
+
+import Foundation
+import ReactiveSwift
+import GoogleSignIn
+import enum Result.NoError
+
+public struct GoogleLoginProviderUser: LoginProviderUser {
+    public let userId: String
+    public let idToken: String
+    public let fullName: String
+    public let givenName: String
+    public let familyName: String
+    public let email: Email?
+    
+    internal init(with user: GIDGoogleUser?) {
+        userId = user?.userID ?? ""
+        idToken = user?.authentication.idToken ?? ""
+        fullName = user?.profile.name ?? ""
+        givenName = user?.profile.givenName ?? ""
+        familyName = user?.profile.familyName ?? ""
+        email = Email(raw: user?.profile.email ?? "")
+        
+    }
+    
+}
+
+public struct GoogleLoginProviderError: LoginProviderError {
+    
+    public let googleError: Error
+    
+    public var localizedMessage: String {
+        return googleError.localizedDescription
+    }
+    
+}
+
+public final class GoogleLoginProvider: NSObject, LoginProvider {
+    
+    public static let name = "Google"
+    
+    public var configuration: GoogleLoginConfiguration = GoogleLoginConfiguration()
+    public let userSignal: Signal<LoginProviderUserType, NoError>
+    public let errorSignal: Signal<LoginProviderErrorType, NoError>
+    
+    fileprivate let userObserver: Observer<LoginProviderUserType, NoError>
+    fileprivate let errorObserver: Observer<LoginProviderErrorType, NoError>
+    
+    public init(with clientId: String) {
+        (userSignal, userObserver) = Signal.pipe()
+        (errorSignal, errorObserver) = Signal.pipe()
+        
+        super.init()
+        // Initialize sign-in
+        GIDSignIn.sharedInstance().delegate = self
+        GIDSignIn.sharedInstance().clientID = clientId //capaz en config
```

Comment:
Yes, I think I would move the clientId  setting to the Configuration, but maybe it's not needed since it's only this, and I don't see what sense it makes to have a provider withuot a client. But either way, remove the comment.